docs/sources/reference/commandline/cli.md

```diff
@@ -599,6 +606,28 @@ repository is used as Dockerfile. Note that you
 can specify an arbitrary Git repository by using the `git://` or `git@`
 schema.
 
+    $ sudo docker build -f Dockerfile.debug .
+
+This will use a file called `Dockerfile.debug` instead of `Dockerfile`
+during the build process. If a `Dockerfile` also exists it will be ignored.
+
+    $ sudo docker build -f dockerfiles/Dockerfile.debug -t myapp_debug .
+    $ sudo docker build -f dockerfiles/Dockerfile.prod  -t myapp_prod .
+
+The above commands will build the current build context twice, once
+using a debug version of a `Dockerfile` and once using a production
+version.
+
+    $ cd /home/me/myapp/some/dir/really/deep
+    $ sudo docker build -f /home/me/myapp/dockerfiles/debug /home/me/myapp
+	$ sudo docker build -f ../../../../dockerfiles/debug /home/me/myapp
```

Comment:
Fixed the indenting - had a tab in there by mistake.

re: -f vs f=  I added a bit more text under the -f/--file section to make it clear that if the value is a relative path then it must be relative to the current dir.